opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -198,3 +198,44 @@ def delete_model(self, model_id: str) -> object:
             method="DELETE",
             url=API_URL,
         )
+
+    def get_profiles(
+        self,
+        node_ids: List[str] = [],
+        model_ids: List[str] = [],
+        task_ids: List[str] = [],
+        return_all_tasks: bool = True,
+        return_all_models: bool = True,
+    ) -> object:
+        """
+        This method retrieves the profile of one or more machine learning nodes, models, or tasks from OpenSearch cluster (using ml commons api)
+
+        :param node_ids: A list of node ids to retrieve profiles for (default: [])
+        :type node_ids: List[str]
+        :param model_ids: A list of model ids to retrieve profiles for (default: [])
+        :type model_ids: List[str]
+        :param task_ids: A list of task ids to retrieve profiles for (default: [])
+        :type task_ids: List[str]
+        :param return_all_tasks: a flag to indicate whether all tasks associated with the profiles should be returned (default: True)
+        :type return_all_tasks: bool
+        :param return_all_models: a flag to indicate whether all models associated with the profiles should be returned (default: True)
+        :type return_all_models: bool
+        :return: returns a json object containing the profile information of the specified nodes, models, or tasks from the OpenSearch cluster
+        :rtype: object
+        """
+
+        API_URL = f"{ML_BASE_URI}/profile"
```

Comment:
Let's make this more robust so that we can also support scenarios like [here](https://opensearch.org/docs/latest/ml-commons-plugin/api/#profile)

1. GET /_plugins/_ml/profile/models --> If we see only one `model_id` is given nothing else, we can invoke this api. Example: `GET /_plugins/_ml/profile/models/model_id`
2. GET /_plugins/_ml/profile/tasks --> if we see only one `task_id` is given nothing else, we can invoke this api. 
Example: `GET /_plugins/_ml/profile/tasks/task_id`
3. Otherwise invoke the base API. Example:

```
GET /_plugins/_ml/profile
{
  "node_ids": ["KzONM8c8T4Od-NoUANQNGg"],
  "return_all_tasks": true,
  "return_all_models": true
}

```

Please let me know if you have any confusion. 

Also don't forget to write test. Let me know if you have any question regarding writing test.

Reply:
1) Endpoint: GET /_plugins/_ml/profile/models/model_id  works fine.
2) Endpoint: GET /_plugins/_ml/profile/tasks/task_id  always returns empty json. So it was deleted. Instead for searching model tasks used API_BODY. 
3) In testing function for traversing over dictionary was added because model_id and task_id in returned json is 4 levels deeper.

Reply:
> 1. GET /_plugins/_ml/profile/models --> If we see only one `model_id` is given nothing else, we can invoke this api. Example: `GET /_plugins/_ml/profile/models/model_id`
> 2. GET /_plugins/_ml/profile/tasks --> if we see only one `task_id` is given nothing else, we can invoke this api.
>    Example: `GET /_plugins/_ml/profile/tasks/task_id`
> 3. Otherwise invoke the base API. Example:

@dhrubo-os : 
For 1, If we implement that, and then user wants all model profile data, its going to cause confusion. Example: User wants all tasks data and so, he doesn't pass model_id nor task_id . How are we going to conclude what user wanted here?

I think, a rather simple interface is to give users to call the respective api's separately.  we should have three separate methods each returning exactly what user is requesting.